zipline/errors.py

```diff
@@ -858,3 +858,14 @@ class IncompatibleTerms(ZiplineError):
         "{term_1} and {term_2} must have the same mask in order to compute "
         "correlations and regressions asset-wise."
     )
+
+
+class BoundColumnInvalidCompare(ZiplineError):
+    """
+    Raised when trying to use BoundColumn with a comparison operator.
+    Frequently occurs when users forget to use '.latest'.
+    """
+    msg = (
+        "Can't compare {column!s} with {other.__class__}.  Did you mean to "
+        "append .latest?"
```

Comment:
"append" was a little unclear to me at first. How about:

```
Can't compare {other.__class__} with {column!s}. Did you mean to compare with `{column!r}.latest`?
```

(Would depend on what the column's repr looks like -- might have to omit the `{column!r}`.)

Reply:
btw the `!s` is the default and technically not needed there

Reply:
On second thought, it might be worth modelling the error message after the default in Python 3.7:

```python
>>> 1 < None
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '<' not supported between instances of 'int' and 'NoneType'
```

So maybe something like:

```
'{op}' not supported between instance of '{other.__class__.__name__}' and '{column}'. Did you mean to compare with '{column}.latest'?
```

(Looks like `BoundColumn.__str__` isn't defined, so it just calls `__repr__`, and so its `!r` and `!s` are equivalent.)

Reply:
Also, might be good to either make `BoundColumnInvalidCompare` inherit from `TypeError`, or just nix the custom class and raise a regular `TypeError` with the custom message.

Reply:
I like your updated error message a lot, but due to the column repr i changed it a little to:
```
"'{op}' not supported between instance of '{compare_type}' and '{column}'. Did you mean use '.latest' with '{column}'?"
```
because putting `.latest` directly on the column repr was a bit confusing.

Reply:
And I got rid of `BoundColumnInvalidCompare` in favor of `TypeError`